dGame/dCinema/Recorder.cpp

```diff
@@ -0,0 +1,1195 @@
+		actor->GetObjectID(),
+		effectID,
+		u"cast",
+		std::to_string(ObjectIDManager::GenerateRandomObjectID())
+	);
+
+	Game::entityManager->SerializeEntity(actor);
+}
+
+void Cinema::Recording::PlayEffectRecord::Serialize(tinyxml2::XMLDocument& document, tinyxml2::XMLElement* parent) {
+	auto* element = document.NewElement("PlayEffectRecord");
+
+	element->SetAttribute("effect", effect.c_str());
+
+	element->SetAttribute("t", m_Delay);
+
+	parent->InsertEndChild(element);
+}
+
+void Cinema::Recording::PlayEffectRecord::Deserialize(tinyxml2::XMLElement* element) {
+	effect = element->Attribute("effect");
+
+	m_Delay = element->DoubleAttribute("t");
+}
+
+void Cinema::Recording::CoroutineRecord::Act(Entity* actor) {
+	
+}
+
+void Cinema::Recording::CoroutineRecord::Serialize(tinyxml2::XMLDocument& document, tinyxml2::XMLElement* parent) {
+	auto* element = document.NewElement("CoroutineRecord");
+
+	for (auto* record : records) {
+		record->Serialize(document, element);
+	}
+
+	element->SetAttribute("t", m_Delay);
+
+	parent->InsertEndChild(element);
+}
+
+void Cinema::Recording::CoroutineRecord::Deserialize(tinyxml2::XMLElement* element) {
+	Recorder::LoadRecords(element, records);
+}
+
+Cinema::Recording::PathFindRecord::PathFindRecord(const NiPoint3& position, float speed) {
+	this->position = position;
+	this->speed = speed;
+}
+
+void Cinema::Recording::PathFindRecord::Act(Entity* actor) {
+}
+
+void Cinema::Recording::PathFindRecord::Serialize(tinyxml2::XMLDocument& document, tinyxml2::XMLElement* parent) {
+	auto* element = document.NewElement("PathFindRecord");
+
+	element->SetAttribute("x", position.x);
+	element->SetAttribute("y", position.y);
+	element->SetAttribute("z", position.z);
+
+	element->SetAttribute("speed", speed);
+
+	element->SetAttribute("t", m_Delay);
+
+	parent->InsertEndChild(element);
+}
+
+void Cinema::Recording::PathFindRecord::Deserialize(tinyxml2::XMLElement* element) {
+	position.x = element->FloatAttribute("x");
+	position.y = element->FloatAttribute("y");
+	position.z = element->FloatAttribute("z");
+	
+	speed = element->FloatAttribute("speed");
+
+	m_Delay = element->DoubleAttribute("t");
+}
+
+Cinema::Recording::CombatAIRecord::CombatAIRecord(bool enabled) {
+	this->enabled = enabled;
+}
+
+void Cinema::Recording::CombatAIRecord::Act(Entity* actor) {
+	auto* baseCombatAIComponent = actor->GetComponent<BaseCombatAIComponent>();
+
+	if (baseCombatAIComponent == nullptr) {
+		return;
+	}
+
+	baseCombatAIComponent->SetDisabled(!enabled);
+}
+
+void Cinema::Recording::CombatAIRecord::Serialize(tinyxml2::XMLDocument& document, tinyxml2::XMLElement* parent) {
+	auto* element = document.NewElement("CombatAIRecord");
+
+	element->SetAttribute("enabled", enabled);
+
+	element->SetAttribute("t", m_Delay);
+
+	parent->InsertEndChild(element);
+}
+
+void Cinema::Recording::CombatAIRecord::Deserialize(tinyxml2::XMLElement* element) {
+	enabled = element->BoolAttribute("enabled");
+
+	m_Delay = element->DoubleAttribute("t");
```

Comment:
May be worth considering not using super short variable names so the xml is more readable and changable outside of code, unless its only intended to be used in code.